### ISSUE TITLE
Add DisallowedCops configuration option to Style/DisableCopsWithinSourceCodeDirective

### ISSUE THREAD

**Issue by corsonknowles** [rubocop/rubocop#15042]
### Before submitting

- [x] I have searched the existing issues and this hasn't been requested yet.

### Cop name

Style/DisableCopsWithinSourceCodeDirective

### Is your feature request related to a problem?

There are a couple of large scale maintenance problems with `Style/DisableCopsWithinSourceCodeDirective` requiring each cop to be specifically allowed:
1. When adding new cops from upstream, it limits our options for handling the rollout in multiple ways --- the most problematic is that a fully accepted cop that turns out to have false positives cannot be easily addressed by developers
2. An opposite problem, for slower rollouts we need to manage every new cop through this config, not just the ones we'd decided to specifically correct. This causes a lot of churn and extra code review gates

The solution that would help us the most is being able to specifically protect certain cops (like `Lint/Void`) using this rule. 

### Describe the solution you'd like

I have a proposed patch to address this:
* https://github.com/rubocop/rubocop/pull/14705

Last time I framed it as relating to the closest issue I found: 
* https://github.com/rubocop/rubocop/issues/14598#issuecomment-3634834832

But we actually have a different set of problems than Rafael with how to use this, so I'm describing those motivations in this issue.

### Code examples

```ruby
Style/DisableCopsWithinSourceCodeDirective:
  Enabled: true
  DisallowedCops: 
    - Lint/Void
```

### Describe alternatives you've considered

Setting aside all the particular issues about additional specs and code review safeguards, we don't have any other good way to deal with the issue of maintaining this rule every time we add new cops. We'd really like to protect a specific list of selected cops (that prevent errors).

### Additional context

Thank you!

**Comment by hammadxcm** [comment]
I've opened a PR to address this: https://github.com/rubocop/rubocop/pull/15050

The implementation adds a `DisallowedCops` configuration option as the inverse of the existing `AllowedCops`. When `DisallowedCops` is set, only directives for the listed cops (and `rubocop:disable all`) are flagged — all other cops can be freely disabled via inline comments.

This is intentionally separate from the discussion in #14598 / #14634 about preventing `Style/DisableCopsWithinSourceCodeDirective` itself from being disabled. The use case here is specifically about protecting a small set of critical cops without needing to allowlist everything else.

@koic @bbatsov @Earlopain Would appreciate your review and feedback on whether this direction works for the project. The change is minimal — a single new config option with the inverse logic of the existing `AllowedCops`, full test coverage, and all CI passing.

cc @corsonknowles (issue author)
